src/builtin.cpp

```diff
@@ -82,8 +81,8 @@
 #include "wutil.h"  // IWYU pragma: keep
 
 static maybe_t<RustBuiltin> try_get_rust_builtin(const wcstring &cmd);
-static proc_status_t builtin_run_rust(parser_t &parser, io_streams_t &streams,
```

Comment:
Is there a reason you aren't using `proc_status_t`? Rust likes strongly types more than C++ does, we want to encourage new types everywhere possible instead of reverting to returning integers.

Reply:
it's because this commit makes `builtin_run_rust` do the same thing as the actual builtin functions which return a `maybe_t<int>`. We might want to change that when we port `builtin_run` to rust but for now there's no reason